addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -223,6 +223,7 @@ function factory(dependencies) {
          */
         _extendMessageDomain(domain) {
             const thread = this.thread;
+            domain = domain.concat([['message_type', '!=', 'user_notification']]);
```

Comment:
I think for channel specifically we might also want to hide `message_type` that are `notification`.
At least they are not counted when checking for `message_unread_counter` (I'm not sure why).

But I'm not sure if the fix here is correct. I think we might be aware of those messages through other means, and in that case we probably still don't want to show them.

Maybe one of the `mail.message` relations of the cache should compute itself to exclude undesired messages? Or having conditions on the viewer template?

Reply:
Initially when messages are loaded on a chatter, user_notification are ignored due to a domain on message_ids in thread model on python side. Since we don't use that field anymore in owl, I think we should have that condition everytime chatter is  involved, so I thought it that way. Is there any use case where user_notification are explicitly loaded from server ?